### PR TITLE
Create android libraries with Dalvik classes, manifest and resources

### LINKED ISSUE #163: Bundle the dalvik classes into aar files
That will also allow bundling in the same file:
- AndroidManifest with required permissions, activities, services, providers
- Custom resources, like xml files
- required dependencies to compile the dalvik classes

Once the aar file is created, Substrate will just need to extract it from the Attach:Android jar, and add it to the Android_project as dependency.

### PR DESCRIPTION
**Pull request by jperedadnr**
This PR fixes #163 

For each service, it builds `META-INF/substrate/dalvik/$serviceName.aar`, a bundle that contains the compiled dalvik classes, a manifest that can contain user permissions, extra activities, services or providers, and possible extra resources.

A follow-up PR in Substrate will replace the use of a fat jar with precompiled classes with the use of these aar files.

### CODE REVIEW

File: gradle/android_project/build.gradle [hunk 1]
```diff
@@ -0,0 +1,20 @@
+buildscript {
+    repositories {
+        mavenCentral()
+        mavenLocal()
+        google()
+        jcenter()
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:4.1.0-alpha09'
```
Comment (johanvos) [inline gradle/android_project/build.gradle:8]: that sounds a scary version (alpha?)

File: gradle/android_project/build.gradle [hunk 2]
```diff
@@ -0,0 +1,20 @@
+buildscript {
+    repositories {
+        mavenCentral()
+        mavenLocal()
```
Comment (johanvos) [inline gradle/android_project/build.gradle]: do we want mavenLocal here?

File: gradle/android_project/gradle.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+org.gradle.daemon=false
+org.gradle.jvmargs=-Xms512M -Xmx4g -XX:MaxPermSize=1024m -XX:MaxMetaspaceSize=1g -Dkotlin.daemon.jvm.options="-Xmx1g"
```
Comment (johanvos) [inline gradle/android_project/gradle.properties]: kotlin?

File: gradle/android_project/library/build.gradle [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+android {
+    compileSdkVersion 28
+
+    defaultConfig {
+        minSdkVersion 21
+        targetSdkVersion 28
+    }
+
+    dependencies {
+        provided fileTree(dir: '../libs', include: '*.jar')
```
Comment (johanvos) [inline gradle/android_project/library/build.gradle:12]: why the relative path here?

File: gradle/mavenPublish.gradle [hunk 5]
```diff
@@ -8,6 +8,12 @@ apply plugin: 'de.marcphilipp.nexus-publish'
 def nativeLib(buildDir, projectDir, service, osname) {
     return tasks.create("nativeLib${service}${osname}") {
         doLast {
+            if (osname == "android") {
+                File aar = aarBuild("$buildDir", "$projectDir", "$service", "$osname")
+                if (aar == null) {
+                    println "Error creating $aar"
```
Comment (johanvos) [inline gradle/mavenPublish.gradle:14]: Will a more detailed error being printed as well?

File: gradle/native-build.gradle [hunk 6]
```diff
@@ -1,7 +1,120 @@
...
     return "/Applications/Xcode.app/Contents/Developer/Platforms/${platform}.platform/Developer/SDKs/${platform}${IOS_VERSION}.sdk";
 }
 
+ext.aarBuild = { buildDir, projectDir, name, os ->
+    def JAVAHOME = System.getenv("JAVA_HOME")
+    def sdk = System.getenv("ANDROID_SDK")
+    if (sdk == null) {
+        throw new GradleException("Error: ANDROID_SDK can't be null")
+    }
+    def androidJar = sdk + "/platforms/android-28/android.jar"
```
Comment (johanvos) [inline gradle/native-build.gradle:13]: here android-28 is used, but on other places I see 21 is the target?

File: gradle/android_project/build.gradle (hunk 1 above)
Comment (jperedadnr) [reply gradle/android_project/build.gradle:8]: Latest release is 4.0.0, but the 4.1.0-alpha version includes some improvements that drop package time more than 50%.

File: gradle/android_project/build.gradle (hunk 2 above)
Comment (jperedadnr) [reply gradle/android_project/build.gradle]: Removed

File: gradle/android_project/gradle.properties (hunk 3 above)
Comment (jperedadnr) [reply gradle/android_project/gradle.properties]: These are default gradle options for an Android project. However I don't think we are going to use Kotlin... Removed

File: gradle/android_project/library/build.gradle (hunk 4 above)
Comment (jperedadnr) [reply gradle/android_project/library/build.gradle:12]: we have a `libs` folder in the root folder, the `library` folder is a subproject

File: gradle/mavenPublish.gradle (hunk 5 above)
Comment (jperedadnr) [reply gradle/mavenPublish.gradle:14]: No, as is. I can add more checks in the `aarBuild` task, and throw an exception if something fails?

File: gradle/native-build.gradle (hunk 6 above)
Comment (jperedadnr) [reply gradle/native-build.gradle:13]: Right, I've removed the `project.properties` file.

**Review by johanvos — APPROVED** [review]

**Pull request merged**